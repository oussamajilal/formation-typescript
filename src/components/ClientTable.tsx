import ClientRow from "./ClientRow";

const ClientTable = () => {
  const clients = [
    {
      id: 1,
      firstName: "Amine",
      lastName: "Saki",
    },
    {
      id: 2,
      firstName: "Ibrahim",
      lastName: "Saki",
    },
    {
      id: 3,
      firstName: "Siham",
      lastName: "Mouli",
    },
    {
      id: 4,
      firstName: "Karam",
      lastName: "Mouli",
    },
  ];

  return (
    <table
      style={{
        backgroundColor: "yellowgreen",
      }}
    >
      <thead>
        <tr>
          <th></th>
          <th>ID</th>
          <th>First Name</th>
          <th>Last Name</th>
          <th>Incremented</th>
          <th>Decremented</th>
          <th>Actions</th>
        </tr>
      </thead>
      <tbody>
        {clients.map((client) => {
          return (
            <ClientRow
              key={client.id}
              id={client.id}
              firstName={client.firstName}
              lastName={client.lastName}
            />
          );
        })}
      </tbody>
    </table>
  );
};

export default ClientTable;
