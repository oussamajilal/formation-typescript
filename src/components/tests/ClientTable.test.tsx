import { render } from "@testing-library/react";
import ClientTable from "../ClientTable";

describe("ClientTable", () => {
  it("should render correctly", () => {
    const dom = render(<ClientTable />);

    expect(dom.asFragment()).toMatchSnapshot();
  });
});
