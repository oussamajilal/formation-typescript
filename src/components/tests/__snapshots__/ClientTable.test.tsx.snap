// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientTable should render correctly 1`] = `
<DocumentFragment>
  <table
    style="background-color: yellowgreen;"
  >
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          First Name
        </th>
        <th>
          Last Name
        </th>
        <th>
          Incremented
        </th>
        <th>
          Decremented
        </th>
        <th>
          Actions
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          Saki
        </td>
        <td>
          Last Name: Saki
        </td>
        <td
          data-testid="inc-cell"
        >
          0
        </td>
        <td
          data-testid="dec-cell"
        >
          0
        </td>
        <td>
          <button
            data-testid="inc-btn"
          >
            Increment
          </button>
          <button
            data-testid="dec-btn"
          >
            Decrement
          </button>
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          Saki
        </td>
        <td>
          Last Name: Saki
        </td>
        <td
          data-testid="inc-cell"
        >
          0
        </td>
        <td
          data-testid="dec-cell"
        >
          0
        </td>
        <td>
          <button
            data-testid="inc-btn"
          >
            Increment
          </button>
          <button
            data-testid="dec-btn"
          >
            Decrement
          </button>
        </td>
      </tr>
      <tr>
        <td>
          3
        </td>
        <td>
          Mouli
        </td>
        <td>
          Last Name: Mouli
        </td>
        <td
          data-testid="inc-cell"
        >
          0
        </td>
        <td
          data-testid="dec-cell"
        >
          0
        </td>
        <td>
          <button
            data-testid="inc-btn"
          >
            Increment
          </button>
          <button
            data-testid="dec-btn"
          >
            Decrement
          </button>
        </td>
      </tr>
      <tr>
        <td>
          4
        </td>
        <td>
          Mouli
        </td>
        <td>
          Last Name: Mouli
        </td>
        <td
          data-testid="inc-cell"
        >
          0
        </td>
        <td
          data-testid="dec-cell"
        >
          0
        </td>
        <td>
          <button
            data-testid="inc-btn"
          >
            Increment
          </button>
          <button
            data-testid="dec-btn"
          >
            Decrement
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
